cli/azd/pkg/commands/pipeline/azdo_provider.go

```diff
@@ -407,6 +402,10 @@ func (p *AzdoScmProvider) gitRepoDetails(ctx context.Context, remoteUrl string)
 	}
 
 	repoDetails := p.getRepoDetails()
+	// Try getting values from the env.
+	// This is a quick shortcut to avoid parsing the remote in detail.
+	// While using the same .env file, the outputs from creating a project and repository
+	// are memorized in .env file
 	if repoDetails.orgName == "" {
 		repoDetails.orgName = p.Env.Values[azdo.AzDoEnvironmentOrgName]
```

Comment:
What happens if it doesn't exist? Does `repoDetails` here allow a `nil` pointer?  You may want to consider checking if the value exists in the env map.

```go
orgName, ok := p.Env.Values[azdo.AzDoEnvironmentOrgName]
```

Then you can check `ok` to determine if it exists.

Reply:
This entire logic of persisting all the azdo configuration in the .env file has a few things to be improved/fixed in next iterations.  @hattan will be working on it.

The change on this PR is only adding a quick patch that fixes https://github.com/Azure/azure-dev/issues/759

That fix can be quite common for customers using `multi .env files` within `azd`. So I made it a blocker and fixed it.

Reply:
To add to this, on a newly created remote the information will always be there since we set it when creating the remote. However, this does pose an issue if some values are present but outdated (such as when running pipeline config multiple times in a row) This PR does in fact fix the issue.

In terms of workflow, repo id, project id, name etc are stored in env for future use. While the logic is there and sound, there are a few edge cases such as the one here. As mentioned, I will look it over to address some edge cases when I'm back.